### DISCUSSION TITLE
.send(.presentCustomAlert, animation: .none) method does not work when present / dismiss fullScreenCover

### DISCUSSION THREAD

**Discussion by JongHoooon** [pointfreeco/swift-composable-architecture#3018] [Q&A]
### Description

I made custom alert. It presented by fullScreenCover. I want to present / dismiss it without animation. 

when present

```swift
Reduce<State, Action> { state, action in
  switch action {

...

    case .someThingHappend:
      return .send(.presentAlert, animation: .none)      //  <-- I want to present without animation

...

    case .presentAlert:
      state.destination = .customAlert(.init(
          title: "alert title",
          message: "alert message"
      ))
      return .none
```

when dismiss

```swift

/* custom alert view */

...

  confirmButton() {
    store.send(.delegate(.confirm), animation: .none)    //  <-- I want to dismiss without animation
  }

...



/* cusomt alert reducer */

enum Action {
    case delegate(Delegate)
    @CasePathable
    enum Delegate: Equatable {
        case confirm
   }
}



/* parent view */

case .destination(.presented(.customAlert(.delegate(.confirm)))):
  state.destination = nil
  return .none

```

### Checklist

- [ ] I have determined whether this bug is also reproducible in a vanilla SwiftUI project.
- [x] If possible, I've reproduced the issue using the `main` branch of this package.
- [X] This issue hasn't been addressed in an [existing GitHub issue](https://github.com/pointfreeco/swift-composable-architecture/issues) or [discussion](https://github.com/pointfreeco/swift-composable-architecture/discussions).

### Expected behavior

I want to present / dismiss full screen cover without animation

### Actual behavior

I want to present / dismiss full screen cover with animation

### The Composable Architecture version information

1.9.3

### Destination operating system

iOS 17

### Xcode version information

15.3

### Swift Compiler version information

```shell
1.90.11.1 Apple Swift version 5.10 (swiftlang-5.10.0.13 clang-1500.3.9.4)
```

**Comment by stephencelis** [comment]
Hi @JongHoooon! The `animation` parameter in TCA is simply a wrapper around `withAnimation`, so the behavior you're seeing is how vanilla SwiftUI behaves, and not a bug with TCA. As such I'm going to convert this to a discussion, but I think you should first try to present/dismiss the full-screen cover without animation in vanilla SwiftUI and that should inform how to do so in TCA.

I believe you can use a `Transaction` that has `disablesAnimation` property set to `true`, but I haven't tried it so I'm not 100% sure of the vanilla solution :smile:

**Reply by JongHoooon** [reply]
disablesAnimation works well but I want to disable only custom alert view. disablesAnimation makes every full screen cover animation.


and I also tried below way..

```swift
/*in reducer body*/

...

case .someAction:
    ...
    return .send(.presnetTestAlert, transaction: .init(animation: .none))     // <--- disable animation not works

...

case .presnetTestAlert:
    state.destination = .alertTest(.init())
    return .none

...
```


It also presents with animation.

**Reply by wickyou23** [reply]
I got the same error, Is there Any new update for this issue?

**Reply by wickyou23** [reply]
It works well when I use `@State` with `fullScreenCover` and `disablesAnimation = true`. But when I send it via TCA, it does not work anymore.

**Reply by mbrandonw** [reply]
Hi @JongHoooon, unfortunately the `transaction` and `animation` arguments of `Effect.send` do not actually work properly, and really can't. We need to deprecate and remove them.

Hi @wickyou23, please start a separate discussion if you are having a problem, and please also include a project that demonstrates the problem.
